File: lab_2/output/input2.py
@dataclass
class User:
    id: float
    name: str
    email: str
    age: float
    isActive: bool

@dataclass
class Product:
    id: float
    title: str
    price: float
    inStock: bool

def createUser2(id: float, name: str, email: str, age: float) -> User:
    return {
    id: id,
    name: name,
    email: email,
    age: age,
    isActive: True
    }

def createProduct(id: float, title: str, price: float) -> Product:
    return {
    id: id,
    title: title,
    price: price,
    inStock: True
    }

def calculateDiscount2(price: float, percent: float) -> float:
    return price - price * percent / 100

isAdult2 = lambda age: age >= 18  # type: (float) -> bool

def printUserName2(user: User) -> None:
    print(user.name)

def printProductTitle(product: Product) -> None:
    print(product.title)

def toggleActive(user: User) -> User:
    user.isActive = not user.isActive
    return user

File: lab_2/input/input2.ts
interface User {
  id: number;
  name: string;
  email: string;
  age: number;
  isActive: boolean;
}

interface Product {
  id: number;
  title: string;
  price: number;
  inStock: boolean;
}

function createUser2(id: number, name: string, email: string, age: number): User {
  return {
    id: id,
    name: name,
    email: email,
    age: age,
    isActive: true
  };
}

function createProduct(id: number, title: string, price: number): Product {
  return {
    id: id,
    title: title,
    price: price,
    inStock: true
  };
}

function calculateDiscount2(price: number, percent: number): number {
  return price - price * percent / 100;
}

const isAdult2 = (age: number): boolean => age >= 18;
const increasePrice2 = (price: number): number => price + 10;

const printUserName2 = (user: User): void => {
  console.log(user.name);
};

const printProductTitle = (product: Product): void => {
  console.log(product.title);
};

function toggleActive(user: User): User {
  user.isActive = !user.isActive;
  return user;
}
